### ISSUE TITLE
Implement `RemoveCodeHash` RPC endpoint

### ISSUE THREAD

**Issue by srdtrk** [cosmos/ibc-go#4951]
## Summary

Currently, there is no way for governance to prevent users from instantiating a wasm client from a previously stored code.

## Problem Definition

You might need to do this if the previously stored code is buggy. The `MigrateContract` RPC  endpoint proposed in https://github.com/cosmos/ibc-go/issues/4925 does not cover this use case as it would only migrate the light client with the given clientID,

## Proposal
1. Add a `RemoveCodeHash` function to `types/wasm.go`
```go
// RemoveCodeHash adds a removes code hash from the list of stored code hashes.
func RemoveCodeHash(ctx sdk.Context, cdc codec.BinaryCodec, codeHash []byte) error {
	codeHashes, err := GetCodeHashes(ctx, cdc)
	if err != nil {
		return err
	}

	// TODO: remove code hash here

	wasmStoreKey := ibcwasm.GetWasmStoreKey()
	store := ctx.KVStore(wasmStoreKey)
	bz, err := cdc.Marshal(&codeHashes)
	if err != nil {
		return err
	}

	store.Set([]byte(KeyCodeHashes), bz)

	return nil
}
```

2. Add a new RPC endpoint called `RemoveCodeHash` to call this the function above. Only authority should be able to call this.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by crodriguezvega** [comment]
Good idea, @srdtrk! I think this is nice to have, but I don't see it as a blocker for v0.1.0 release. I could see this added in a future release, since I don't expect we will encounter so soon the situation that we need to migrate contracts and we should remove the code hash of a buggy contract. What do you think?

**Comment by srdtrk** [comment]
Agreed. I think given that it is a relatively small addition. I think I can implement this easily before we mark `v0.1.0`. I'll mark it as a nice to have then, and see if we have time to implement it before `v0.1`

**Comment by crodriguezvega** [comment]
Closed by #5006
